src/py/flwr/server/state/sqlite_state.py

```diff
@@ -0,0 +1,582 @@
+        """Create tables if they don't exist yet."""
+        self.conn = sqlite3.connect(self.database_path)
+        self.conn.row_factory = dict_factory
+        self.conn.set_trace_callback(lambda query: log(1, query))
+        cur = self.conn.cursor()
+
+        # Create each table if not exists queries
+        cur.execute(SQL_CREATE_TABLE_TASK_INS)
+        cur.execute(SQL_CREATE_TABLE_TASK_RES)
+        cur.execute(SQL_CREATE_TABLE_NODE)
+
+        res = cur.execute("SELECT name FROM sqlite_schema;")
+
+        return res.fetchall()
+
+    def query(
+        self,
+        query: str,
+        data: Optional[Union[List[DictOrTuple], DictOrTuple]] = None,
+    ) -> List[Dict[str, Any]]:
+        """Run SQLQuery."""
+        if self.conn is None:
+            raise Exception("State is not initialized.")
+
+        if data is None:
+            data = []
+
+        # Cleanup whitespace to make the logs nicer
+        query = re.sub(r"\s+", " ", query)
+
+        try:
+            with self.conn:
+                if (
+                    len(data) > 0
+                    # pylint: disable-next=C0123
+                    and (type(data) == tuple or type(data) == list)
+                    # pylint: disable-next=C0123
+                    and (type(data[0]) == tuple or type(data[0]) == dict)
+                ):
+                    rows = self.conn.executemany(query, data)
+                else:
+                    rows = self.conn.execute(query, data)
+
+                # Extract results before commiting to support
+                #   INSERT/UPDATE ... RETURNING
+                # style queries
+                result = rows.fetchall()
+        except KeyError as exc:
+            log(ERROR, {"query": query, "data": data, "exception": exc})
+
+        return result
+
+    def store_task_ins(self, task_ins: TaskIns) -> Optional[UUID]:
+        """Store one TaskIns.
+
+        Usually, the Driver API calls this to schedule instructions.
+
+        Stores the value of the task_ins in the state and, if successful, returns the
+        task_id (UUID) of the task_ins. If, for any reason, storing the task_ins fails,
+        `None` is returned.
+
+        Constraints
+        -----------
+        If `task_ins.task.consumer.anonymous` is `True`, then
+        `task_ins.task.consumer.node_id` MUST NOT be set (equal 0). Any implemenation
+        may just override it with zero instead of validating.
+
+        If `task_ins.task.consumer.anonymous` is `False`, then
+        `task_ins.task.consumer.node_id` MUST be set (not 0)
+        """
+        # Validate task
+        errors = validate_task_ins_or_res(task_ins)
+        if any(errors):
+            log(ERROR, errors)
+            return None
+
+        # Create and set task_id
+        task_id = uuid4()
+        task_ins.task_id = str(task_id)
+
+        # Set created_at
+        created_at: datetime = _now()
+        task_ins.task.created_at = created_at.isoformat()
+
+        # Set ttl
```

Comment:
```suggestion
        # Create and set ttl
```

Reply:
Making it consistent with `in_memory_state`